x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -40,6 +44,12 @@ import {
 } from './template';
 import { buildDefaultSettings } from './default_settings';
 
+const mergeWithArrayConcat = (a: any, b: any) =>
+  mergeWith(a, b, (val1, val2) => {
+    if (isArray(val1)) {
```

Comment:
what happens when `val1` is not an array? shouldn't `a` and `b` have type `Array` instead?

Reply:
When val1 isn't an array we revert to the default merging behaviour of lodash mergeWith, I will add a comment here